### ISSUE TITLE
contributing back to OpenPIV

### ISSUE THREAD

**Issue by alexlib** [Stergrim/PIV-Particle-Image-Velocimetry#1]
Dear @Stergrim, This is one of the most interesting derivatives of OpenPIV. How could you extend OpenPIV with your developments?

**Comment by Stergrim** [comment]
Dear @alexlib, to answer your question briefly, there is no way, in any way, I could expand OpenPIV with my own developments. I will answer in more detail below.
My PIV project is just a modified version of what I did at university. I'm not interested in that right now. I gave up my scientific work. I see further development of the PIV field only with the use of neural networks, since they have higher processing speed and accuracy (for example, PIV-LiteFlowNet-en). The disadvantage of neural networks is their narrow focus, that is, the lack of flexible configuration for various cases. But to solve this problem and develop neural networks, I do not have the necessary knowledge and skills. In my master's thesis, there is a comparison of a neural network with a self-written PIV method. If you are interested in this, I can tell you a little more about what disadvantages I have found in neural networks and how I see their solution.
Have a nice day.

**Comment by alexlib** [comment]
Thanks for a prompt answer. I would appreciate a chat about the comparison and your view on neural networks and PIV

shall we get on zoom? could you share your comparison ? 
I don't want to run all the NN again and would like to learn from your experience.

Alex 
alex dot liberzon at gmail dot com

**Comment by Stergrim** [comment]
I think it makes no sense to switch to zoom, since my knowledge of English is not enough for speaking. I will prepare an answer comparing neural networks and PIV, but this experience is limited to 2021. A detailed answer will take some time, more than one day. If you want to get acquainted with my research now, then my master's thesis is in the attachment. Paragraph 1.3 is an overview of articles on machine learning, and paragraphs 2.3 and 2.4 are the application and study of the limitations of neural networks.
https://github.com/Stergrim/Recalibration-of-a-stereo-pair-based-on-a-reprojection-error/blob/main/demos/DiplomMaster.pdf

**Comment by Stergrim** [comment]
I will not describe the theory of neural networks and their structures.

At the time of writing the master's thesis, the most suitable open source neural network was LiteFlowNet and its version trained for PIV tasks (PIV-LiteFlowNet).
I found such a graph from the publication [80]*, which presents a comparison of neural networks. 
![31](https://github.com/user-attachments/assets/2bee544c-52bb-48be-aa84-3b2dcc2a2eac)
DCVNet has not found open source code at the time of writing the work.
When reviewing neural networks, I was mainly interested in neural networks of a wide profile, that is, working with various types of flow, and not just with some specific type of flow.

I will describe the features and limitations that I noticed when using LiteFlowNet and PIV-LiteFlowNet.
1) Maximum estimated offset:
LiteFlowNet – 80-90 pixels;
PIV-LiteFlowNet – 12-13 pixels.
For LiteFlowNet, this is most likely caused by its number of layers (levels). And for PIV-LiteFlowNet, the restriction is caused by a training sample, on which the maximum offset is 11 pixels.
2) The difference in accuracy with a dark and light background, i.e. inversion of images:
LiteFlowNet – shows the best results on a light background, i.e. a white background with black dots. And PIV-LiteFlowNet is the opposite.
This is most likely caused by a training sample.
A little more about paragraphs 1 and 2 in the master's thesis, paragraph 2.4.4.
These are the main features that I have identified: the maximum of the estimated bias and the distinctive features of the training sample.

I would like to note that these neural networks were not originally created for PIV tasks. They have a broader application profile. In order to adapt these neural networks for their tasks, research groups mainly changed only the training sample to the PIV dataset.
The reason for this approach is that it is very difficult to develop a good neural network and the task is beyond the power of a research group in which neural network development is not the main direction and specialty. Therefore, it is easier to form a training sample.

I see 3 options to develop this direction:
1. Develop your own neural network for PIV tasks. It's very difficult.
2. Expand the training sample. The best option is the data obtained in laboratory experiments. It is possible to model the data, but this is done by separate laboratories, since complex modeling using the Navier-Stokes equations. Difficult.
3. Search for existing neural networks from related fields, train them on an existing PIV sample and see the results. Normal.

To partially solve the shortage of training samples, there is an UnLiteFlowNet neural network. I have not studied this topic in depth, but learning with uncontrolled sampling is very interesting.

I can't tell you which neural networks are the best right now, since this review was completed at the end of 2021, but I think it's worth looking for information about DCVNet.

*[80] - Huaizu Jiang Caltech and Erik Learned-Miller UMass Amherst, "DCVNet: Dilated Cost Volume Networks for Fast Optical Flow", arXiv:2103.17271 (2021).
